### PR TITLE
Migrate image models to cpp_server (SDXL)

### LINKED ISSUE #2375: Move image models to C++ server

### PR DESCRIPTION
**Pull request by ztorlakTT**
Migrates SDXL image models (text-to-image, image-to-image, edit/inpaint) from the Python `tt-media-server` into the C++ `cpp_server`, behind the same OpenAI-compatible endpoints. Selected at runtime via `MODEL_SERVICE=image` and `MODEL_RUNNER_TYPE=tt_sdxl_{generate,image_to_image,edit}`.

- **New endpoints** (controller `ImageController`): `POST /v1/images/generations`, `POST /v1/images/image-to-image`, `POST /v1/images/edits`. Only the route matching the active `MODEL_RUNNER_TYPE` is registered.
- **Runner family for in-process media work**: `IMediaRunner<Request, Response>` alongside the existing IPC-loop `IRunner`, sharing a common `IRunnerBase` (warmup/stop/runnerType). SDXL pipelines are driven from C++ via pybind11, with all Python access guarded by the GIL.
- **Single `RunnerRegistry`** that handles both families: `registerIpcRunner` / `createIpc` for worker-process runners (LLM, embedding) with MOCK fallback, and `registerMediaRunner` / `createMedia<Runner>()` for direct-call media runners with type-safe downcasting.
- **Shared media config** (`MediaRunnerConfigBase`): `max_batch_size`, `device_mesh_shape`, `is_galaxy`, `model_weights_path`, `weights_distribution_timeout_seconds`. Read once via `readMediaRunnerConfig`, ready for upcoming audio/TTS/video runners.
- **Generic sync dispatch helper** (`dispatchJsonRequest`) used by `ImageController` and reusable by future media controllers — handles JSON validation, model-readiness, request parsing, thread-pool offload, and OpenAI-style error mapping (400 / 429 / 500 / 503).
- **Plumbing cleanup**: `runner_factory` renamed to `ipc_runner_factory` to reflect its IPC-only role; `runner_interface.hpp` → `ipc_runner.hpp`; `IService` lifecycle hooks (`warmup`, `stop`, `runnerType`) standardized on `IRunnerBase`. `RouteRegistry` only carries routes for the active service.

CI runs:
- SDXL N150: https://github.com/tenstorrent/tt-shield/actions/runs/25909766416
- SDXL P300x2: https://github.com/tenstorrent/tt-shield/actions/runs/25909734330
- SDXL galaxy: https://github.com/tenstorrent/tt-shield/actions/runs/25909609318
- SDXL Img2Img galaxy: https://github.com/tenstorrent/tt-shield/actions/runs/25909583050
- SDXL Inpainting galaxy: https://github.com/tenstorrent/tt-shield/actions/runs/25909561489

### CODE REVIEW

File: tt-media-server/cpp_server/include/domain/image/image_response.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  Json::Value toOpenaiJson() const {
+    Json::Value json;
+    Json::Value imagesArr(Json::arrayValue);
+    for (const auto& img : images) imagesArr.append(img);
+    json["images"] = std::move(imagesArr);
+    if (generation_time_seconds > 0.0) {
+      json["generation_time"] = generation_time_seconds;
+    }
+    return json;
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/include/domain/image/image_response.hpp:38]: Nitpick: would it make sense to split domains per types? i.e. image/text/...
This might grow quite a lot

File: tt-media-server/cpp_server/include/runners/sdxl/sdxl_runner.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      int neededPadding) const;
+
+  struct PromptPack {
+    std::vector<std::string> prompts;
+    std::optional<std::vector<std::string>> negative_prompts;
+    std::optional<std::vector<std::string>> prompts_2;
+    std::optional<std::string> negative_prompt_2;
+    int needed_padding = 0;
+  };
+  /** Pad prompts to max_batch_size. */
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/include/runners/sdxl/sdxl_runner.hpp]: Do we really have this?

File: tt-media-server/cpp_server/include/runners/media_runner.hpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+#pragma once
+
+#include "runners/runner_base.hpp"
+
+namespace tt::runners {
+
+/** Direct-call runner owned by an in-process service; the service maps
+ *  exceptions thrown from `run()` to error responses. */
+template <typename Request, typename Response>
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/include/runners/media_runner.hpp:12]: I'm not 100% clear on this. Would this be a difference on scheduling policy between media runner and ipc runner?

Also media might not be the best term since whisper might have streaming

File: tt-media-server/cpp_server/src/api/image_controller.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+ImageController::~ImageController() = default;
+
+void ImageController::generate(
+    const drogon::HttpRequestPtr& req,
+    std::function<void(const drogon::HttpResponsePtr&)>&& callback) {
+  dispatchJsonRequest<tt::services::ImageService,
+                      tt::domain::ImageGenerateRequest>(
+      req, std::move(callback), service_, request_counter_, "ImageController",
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/src/api/image_controller.cpp:36]: How do we use request counter here?

File: tt-media-server/cpp_server/src/runners/sdxl/sdxl_runner.cpp [hunk 5]
```diff
@@ -0,0 +1,742 @@
...
+      work();
+      promise->set_value();
+    } catch (...) {
+      try {
+        promise->set_exception(std::current_exception());
+      } catch (...) {
+      }
+    }
+  });
+  worker.detach();
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/src/runners/sdxl/sdxl_runner.cpp]: Not sure is this optimal, maybe we can use the same approach as prefill runner?

File: tt-media-server/cpp_server/src/runners/sdxl/sdxl_runner.cpp [hunk 6]
```diff
@@ -0,0 +1,742 @@
...
+    throw std::runtime_error(std::string("[SDXL] inference failed: ") +
+                             e.what());
+  }
+}
+
+// ---------------------------------------------------------------------------
+// SDXLGenerateRunner
+// ---------------------------------------------------------------------------
+
+SDXLGenerateRunner::SDXLGenerateRunner(const config::ImageConfig& config)
```
Comment (idjuricTT) [inline tt-media-server/cpp_server/src/runners/sdxl/sdxl_runner.cpp]: Nitpick: maybe we can have separate files for this

**Review by idjuricTT — APPROVED** [review]

File: tt-media-server/cpp_server/include/runners/media_runner.hpp (hunk 3 above)
Comment (ztorlakTT) [reply tt-media-server/cpp_server/include/runners/media_runner.hpp:12]: This runner is called directly from the service + the run method needs a request as a difference from IRunner. I would handle this in a separate PR because this one is already too big.
I also agree about the naming, do you have any suggestions?

File: tt-media-server/cpp_server/src/api/image_controller.cpp (hunk 4 above)
Comment (ztorlakTT) [reply tt-media-server/cpp_server/src/api/image_controller.cpp:36]: We use it for logging, for tracking image requests. We also have it in the embedding controller

File: tt-media-server/cpp_server/include/runners/sdxl/sdxl_runner.hpp (hunk 2 above)
Comment (ztorlakTT) [reply tt-media-server/cpp_server/include/runners/sdxl/sdxl_runner.hpp]: If you mean padding up to max_batch_size, then yes

**Review by idjuricTT — APPROVED** [review]
